gremlin-go/driver/client.go

```diff
@@ -71,25 +73,51 @@ func NewClient(url string, configurations ...func(settings *ClientSettings)) (*C
 		transporterType: settings.TransporterType,
 		connection:      conn,
 	}
+	// TODO: PoolSize must be 1 on Session mode
 	return client, nil
 }
 
 // Close closes the client via connection.
 func (client *Client) Close() error {
-	return client.connection.close()
+	// If it is a Session, call closeSession
+	if client.Session != "" {
+		_, err := client.closeSession()
+		if err != nil {
+			return err
+		}
+	}
+	client.logHandler.logf(Info, "Closing Client with url %s", client.url)
+	err := client.connection.close()
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 // Submit submits a Gremlin script to the server and returns a ResultSet.
 func (client *Client) Submit(traversalString string) (ResultSet, error) {
-	// TODO AN-982: Obtain connection from pool of connections held by the client.
 	client.logHandler.logf(Debug, submitStartedString, traversalString)
 	request := makeStringRequest(traversalString, client.traversalSource)
+	// TODO: Add bindings to request. request.args['bindings'] = bindings
+	if client.Session != "" {
+		request.args["session"] = client.Session
+		request.processor = "session"
+	}
```

Comment:
Since this is shared functionality for modifying a request as below, let's make a helper method in `request.go`